packages/scandipwa/src/component/Router/Router.container.js

```diff
@@ -137,6 +148,10 @@ export class RouterContainer extends PureComponent {
 
     componentDidMount() {
         window.addEventListener('resize', this.handleResize);
+
+        history.listen(this.handleRouteChange([
+            this.updateCartOnRouteChange
+        ]));
```

Comment:
But there is just one callback and unlikely we will add another one.
Please reduce the logic here. It's too complex.

Reply:
https://github.com/scandipwa/scandipwa/issues/4971

I will push this in a few minutes :D